test/finite_differencing.jl

```diff
@@ -1,58 +1,78 @@
+        # # Check that finite differencing yields approximately correct results for vectors.
         N = 3
-        ȳ, x, v = ones.((N, N, N))
-        @test Nabla.compute_Dv(foo, ȳ, x, v) == sum(ȳ .* (UniformScaling(10) * v))
-        @test Nabla.approximate_Dv(foo, ȳ, x, v) == sum(ȳ .* (UniformScaling(10) * v))
-        ȳ = randn(N)
-        @test Nabla.compute_Dv(foo, ȳ, x, v) == sum(ȳ .* (UniformScaling(10) * v))
-        @test Nabla.approximate_Dv(foo, ȳ, x, v) == sum(ȳ .* (UniformScaling(10) * v))
+        z̄, x, y, vx, vy = ones.((N, N, N, N, N))
+        @test compute_Dv(foo, z̄, (x, y), (vx, vy)) ≈
+            sum(z̄ .* (UniformScaling(10) * vx)) + sum(z̄ .* (UniformScaling(11) * vy))
+        @test approximate_Dv(foo, z̄, (x, y), (vx, vy)) ≈
+            sum(z̄ .* (UniformScaling(10) * vx)) + sum(z̄ .* (UniformScaling(11) * vy))
+        z̄ = randn(N)
+        @test compute_Dv(foo, z̄, (x, y), (vx, vy)) ≈
+            sum(z̄ .* (UniformScaling(10) * vx)) + sum(z̄ .* (UniformScaling(11) * vy))
+        @test approximate_Dv(foo, z̄, (x, y), (vx, vy)) ≈
+            sum(z̄ .* (UniformScaling(10) * vx)) + sum(z̄ .* (UniformScaling(11) * vy))
         v = randn(N)
-        @test Nabla.compute_Dv(foo, ȳ, x, v) ≈ sum(ȳ .* (UniformScaling(10) * v))
-        @test Nabla.approximate_Dv(foo, ȳ, x, v) ≈ sum(ȳ .* (UniformScaling(10) * v))
+        @test compute_Dv(foo, z̄, (x, y), (vx, vy)) ≈
+            sum(z̄ .* (UniformScaling(10) * vx)) + sum(z̄ .* (UniformScaling(11) * vy))
+        @test approximate_Dv(foo, z̄, (x, y), (vx, vy)) ≈
+            sum(z̄ .* (UniformScaling(10) * vx)) + sum(z̄ .* (UniformScaling(11) * vy))
 
         # Check that finite differencing yields approximately correct results for matrices.
         M, N = 3, 2
-        Ȳ, X, V = ones.((M, M, M), (N, N, N))
-        @test Nabla.approximate_Dv(foo, Ȳ, X, V) == Nabla.compute_Dv(foo, Ȳ, X, V)
+        Z̄, X, Y, VX, VY = ones.((M, M, M, M, M), (N, N, N, N, N))
+        @test approximate_Dv(foo, Z̄, (X, Y), (VX, VY)) ≈
+            compute_Dv(foo, Z̄, (X, Y), (VX, VY))
         Ȳ = randn(M, N)
-        @test Nabla.approximate_Dv(foo, Ȳ, X, V) == Nabla.compute_Dv(foo, Ȳ, X, V)
+        @test approximate_Dv(foo, Z̄, (X, Y), (VX, VY)) ≈
+            compute_Dv(foo, Z̄, (X, Y), (VX, VY))
     end
+end
 
-    function print_tol_err(f, ȳ, x::T, v::T, err_abs::∇Real, err_rel::∇Real) where T<:ArrayOr∇Real
-        println("Large error found in sensitivity for function $f at input")
-        println(x)
-        println("in direction")
-        println(v)
-        println("err_abs = $err_abs, err_rel = $err_rel")
-        throw(error("Large error found in sensitivity."))
-    end
+function print_tol_err(f, ȳ, x::T, v::T, err_abs::∇Real, err_rel::∇Real) where T<:ArrayOr∇Real
+    println("Large error found in sensitivity for function $f at input")
+    println(x)
+    println("in direction")
+    println(v)
+    println("err_abs = $err_abs, err_rel = $err_rel")
```

Comment:
Consider formatting `err_abs` and `err_rel`for readability, and consider also printing the maximum errors.